File: app/src/main/java/com/gadgetsfolk/kidsgk/model/QuizType.java
package com.gadgetsfolk.kidsgk.model;

import android.os.Parcel;
import android.os.Parcelable;

import androidx.annotation.Keep;

@Keep
public class QuizType implements Parcelable {
    private String title, sub_title;
    private String img_url;
    private String doc_id;

    public String getDoc_id() {
        return doc_id;
    }

    public void setDoc_id(String doc_id) {
        this.doc_id = doc_id;
    }

    public QuizType(){}

    private QuizType(Parcel parcel) {
        title = parcel.readString();
        sub_title = parcel.readString();
        img_url = parcel.readString();
        doc_id = parcel.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(title);
        parcel.writeString(sub_title);
        parcel.writeString(img_url);
        parcel.writeString(doc_id);
    }

    public static final Creator<QuizType> CREATOR = new Creator<QuizType>() {
        @Override
        public QuizType createFromParcel(Parcel parcel) {
            return new QuizType(parcel);
        }

        @Override
        public QuizType[] newArray(int i) {
            return new QuizType[i];
        }
    };

    public String getImg_url() {
        return img_url;
    }

    public void setImg_url(String img_url) {
        this.img_url = img_url;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getSub_title() {
        return sub_title;
    }

    public void setSub_title(String sub_title) {
        this.sub_title = sub_title;
    }

}

File: app/src/main/java/com/gadgetsfolk/kidsgk/model/QuizType.kt
package com.gadgetsfolk.kidsgk.model

import android.os.Parcel
import android.os.Parcelable
import android.os.Parcelable.Creator
import androidx.annotation.Keep

@Keep
class QuizType : Parcelable {
    var title: String? = null
    var sub_title: String? = null
    var img_url: String? = null
    var doc_id: String? = null

    constructor() {}
    private constructor(parcel: Parcel) {
        title = parcel.readString()
        sub_title = parcel.readString()
        img_url = parcel.readString()
        doc_id = parcel.readString()
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeString(title)
        parcel.writeString(sub_title)
        parcel.writeString(img_url)
        parcel.writeString(doc_id)
    }

    companion object CREATOR : Creator<QuizType> {
        override fun createFromParcel(parcel: Parcel): QuizType {
            return QuizType(parcel)
        }

        override fun newArray(size: Int): Array<QuizType?> {
            return arrayOfNulls(size)
        }
    }
}